addons/mail/models/mail_thread.py

```diff
@@ -1301,6 +1294,34 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
             (email_from, email_to, message_id)
         )
 
+    @api.multi
+    def _message_find_user_id(self, email_from, alias=None, fallback_model=False):
+        email_from = (tools.email_split(email_from) or [''])[0]
```

Comment:
Just for the record, you can now use : 
`email_from = tools.email_normalize(email_from)`
and search on user.email_normalized to be sure that the email_from and the user's email correspond

Reply:
@dbeguin , 👍  done the changes. Thanks